src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] output = readFile(fromFileName);
+        String[] result = generateReport(output);
+        writeToFile(result, toFileName);
+    }
+
+    public static String[] readFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int linesCount = 0;
+            while (reader.readLine() != null) {
+                linesCount++;
+            }
+            String[] output = new String[linesCount];
+            try (BufferedReader readerAgain = new BufferedReader(new FileReader(fromFileName))) {
+                String line;
+                int index = 0;
+                while ((line = readerAgain.readLine()) != null) {
+                    output[index++] = line.trim();
+                }
+            }
+            return output;
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file.", e);
+        }
+    }
```

Comment:
The method 'readFile' should not be static as it is part of the object's behavior. Additionally, this method is opening a connection to the file twice, which is inefficient. Consider reading the file and storing its content in a single pass without reopening the file.